lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,7 @@ def word_symbol_pair?(pair)
           if key.sym_type?
             sym_name = key.loc.expression.source
 
-            sym_name !~ /\A:["']/
+            sym_name !~ /\A:["']/ && !sym_name.end_with?('=')
```

Comment:
I actually think we should compare the symbols source (e.g. :"something" to the symbol as presented in the node). Quotes are stripped for symbols that don't need them, so this would make the code simpler and more robust.